dev/com.ibm.ws.security.openidconnect.clients.common/test/com/ibm/ws/security/openidconnect/client/jose4j/util/Jose4jUtilTest.java

```diff
@@ -421,6 +425,40 @@ public void testGetVerifyKey_getKeyFromJwkUrl() throws Exception {
         }
     }
 
+    /************************************** getThirdPartyClaims **************************************/
+
+    @Test
+    public void testGetThirdPartyClaims() throws Exception {
+        Set<String> allowedThirdPartyTokenClaims = new HashSet<String>();
+        allowedThirdPartyTokenClaims.add("test1");
+        allowedThirdPartyTokenClaims.add("test4");
+
+        Map<String, Object> thirdPartyClaims = Jose4jUtil.getThirdPartyClaims(THIRD_PARTY_TOKEN, allowedThirdPartyTokenClaims);
+
+        assertTrue("Expected third-party claims map to contain claim 'test1'.", thirdPartyClaims.containsKey("test1"));
+        assertTrue("Expected third-party claims map to contain claim 'test4'.", thirdPartyClaims.containsKey("test4"));
+    }
+
+    @Test
+    public void testGetThirdPartyClaims_noClaimsSpecified() throws Exception {
+        Set<String> allowedThirdPartyTokenClaims = new HashSet<String>();
+
+        Map<String, Object> thirdPartyClaims = Jose4jUtil.getThirdPartyClaims(THIRD_PARTY_TOKEN, allowedThirdPartyTokenClaims);
+
+        assertTrue("Expected third-party claims map to be empty.", thirdPartyClaims.isEmpty());
+    }
+
+    @Test
+    public void testGetThirdPartyClaims_someClaimsNotSpecified() throws Exception {
+        Set<String> allowedThirdPartyTokenClaims = new HashSet<String>();
+        allowedThirdPartyTokenClaims.add("test1");
+
+        Map<String, Object> thirdPartyClaims = Jose4jUtil.getThirdPartyClaims(THIRD_PARTY_TOKEN, allowedThirdPartyTokenClaims);
+
+        assertTrue("Expected third-party claims map to contain claim 'test1'.", thirdPartyClaims.containsKey("test1"));
+        assertFalse("Expected third-party claims map to not contain claim 'test4'.", thirdPartyClaims.containsKey("test4"));
+    }
+
```

Comment:
Another test that I think would be good to add that I believe Chris had mentioned earlier would be to have a claim in the allowed third party token claims list that _isn't_ in the token.